### DISCUSSION TITLE
System fields are not getting added to records

### DISCUSSION THREAD

**Discussion by shiva-rakshith** [Sunbird-RC/community#355] [Issues]
Hi, We are using sunbird-rc v0.0.7 and added the below system fields to an existing schema. When a new record is created, following fields are not getting added. 

```
"systemFields": [
        "_osCreatedAt",
        "_osUpdatedAt",
        "_osCreatedBy",
        "_osUpdatedBy"
        ]
```        

cc: @maheshkumargangula

**Comment by tejash-jl** [comment]
@shiva-rakshith remove the underscore in the beginning.

**Reply by tejash-jl** [reply]
also, let me know where you got theses values, ill get it fixed

**Reply by shiva-rakshith** [reply]
Got those fields from here: https://docs.sunbirdrc.dev/developer-documentation/creating-your-own-schemas

**Reply by tejash-jl** [reply]
ok, i updated it

**Reply by AbhiGaddi** [reply]
Hi @tejash-jl 
Thanks for the help,
we are able to get the System fields, but they are adding to every entity in the schema, can we restrict only to the parent entity.
Example: If USER is an entity. That contains address as a child entity, system fields are adding to both address   and USER entity.

**Comment by AbhiGaddi** [comment]
@tejash-jl 

we are getting empty values for the createdBy and updatedBy fields.
Because of setting userID as empty in the invite API.
[Line setting empty](https://github.com/Sunbird-RC/sunbird-rc-core/blob/86fc18cff3df326bc8bf84dc398dbc4f2a3d64eb/java/registry/src/main/java/dev/sunbirdrc/registry/controller/RegistryEntityController.java#L86)
is this expected behaviour, or do we have any configuration to override userID ?

**Reply by tejash-jl** [reply]
Yes, it is expected behaviour, invite API will be used to invite another user to the registry by providing email/mobile.
If you need the createdBy & updatedBy then POST api/v1/<Entity> can be used,
https://docs.sunbirdrc.dev/api-reference/registry/create-an-entity
